tests/test_middleware.py

```diff
@@ -574,6 +518,63 @@ def test_histogram(self, client):
             in metrics
         )
 
+    def test_custom_root_path(self, testapp):
+        """test that custom root_path does not affect the path grouping"""
+
+        client = TestClient(testapp(skip_paths=["/health"]), root_path="/api")
+
+        client.get("/200/111")
+        client.get("/500/1111")
+        client.get("/404/123")
+
+        client.get("/api/200/111")
```

Comment:
Curious, does starlette's router match both `/200/111` and `/api/200/111` (with or without the `/api` prefix) to the  same endpoint if root_path is set?

Thanks for yet another large contribution @dolfinus

Reply:
> Curious, does starlette's router match both /200/111 and /api/200/111 (with or without the /api prefix) to the same endpoint if root_path is set?

Yes. But not for Mount and StaticFiles, for some reason